src/main/java/org/jenkinsci/plugins/workflow/support/steps/ExecutorStepExecution.java

```diff
@@ -376,6 +357,20 @@ private Object readResolve() {
             return this;
         }
 
+        /**
+         * We cannot keep {@link ExecutorStepExecution} as a serial field of {@link PlaceholderTask}
+         * since it could not be serialized via XStream in {@link Queue}.
+         * Instead we keep only {@link #context} and look up the execution as needed.
+         */
+        private void withExecution(Consumer<ExecutorStepExecution> executionCallback) {
+            StepExecution.applyAll(ExecutorStepExecution.class, execution -> {
```

Comment:
This iterates over all running steps in all running Pipelines, right? Why not look up the `FlowExecution` from `context` instead so that we can call `FlowExecution.getCurrentExecutions` directly to only consider steps for this build? Maybe just because this is expected to run infrequently and so performance is not a concern?

Reply:
Cannot recall whether I tried that. Can check if it works.